File: src/com/mycompany/entities/Produit.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entities;

/**
 *
 * @author 21656
 */
public class Produit {
   private int id_produit;
    private String nom_produit;
    private String description_produit;
    private float prix_produit;
    private String image_produit;
    private Categorie id_categorie;
    private user user;
    public Produit() {
    }

    public Produit(int id_produit, String nom_produit, String description_produit, float prix_produit) {
        this.id_produit = id_produit;
        this.nom_produit = nom_produit;
        this.description_produit = description_produit;
        this.prix_produit = prix_produit;
    }

    public Produit(int id_produit, String nom_produit, String description_produit, float prix_produit, String image_produit,Categorie id_categorie) {
        this.id_produit = id_produit;
        this.nom_produit = nom_produit;
        this.description_produit = description_produit;
        this.prix_produit = prix_produit;
        this.image_produit = image_produit;
        
    }

    public Produit(String nom_produit, String description_produit, float prix_produit, String image_produit, Categorie id_categorie, user user) {
        this.nom_produit = nom_produit;
        this.description_produit = description_produit;
        this.prix_produit = prix_produit;
        this.image_produit = image_produit;
        this.id_categorie = id_categorie;
        this.user = user;
    }

    public Produit(int id_produit, String nom_produit, String description_produit, float prix_produit, String image_produit, Categorie id_categorie, user user) {
        this.id_produit = id_produit;
        this.nom_produit = nom_produit;
        this.description_produit = description_produit;
        this.prix_produit = prix_produit;
        this.image_produit = image_produit;
        this.id_categorie = id_categorie;
        this.user = user;
    }

    

    public int getId_produit() {
        return id_produit;
    }

    public String getNom_produit() {
        return nom_produit;
    }

    public String getDescription_produit() {
        return description_produit;
    }

    public float getPrix_produit() {
        return prix_produit;
    }

    public String getImage_produit() {
        return image_produit;
    }

    public Categorie getId_categorie() {
        return id_categorie;
    }

    public void setUser(user user) {
        this.user = user;
    }

    public user getUser() {
        return user;
    }
    

    

    

    public void setId_produit(int id_produit) {
        this.id_produit = id_produit;
    }

    public void setNom_produit(String nom_produit) {
        this.nom_produit = nom_produit;
    }

    public void setDescription_produit(String description_produit) {
        this.description_produit = description_produit;
    }

    public void setPrix_produit(float prix_produit) {
        this.prix_produit = prix_produit;
    }

    public void setImage_produit(String image_produit) {
        this.image_produit = image_produit;
    }

    public void setId_categorie(Categorie id_categorie) {
        this.id_categorie = id_categorie;
    }

    @Override
    public String toString() {
        return "Produit{" + "id_produit=" + id_produit + ", nom_produit=" + nom_produit + ", description_produit=" + description_produit + ", prix_produit=" + prix_produit + ", image_produit=" + image_produit + ", id_categorie=" + id_categorie + ", user=" + user + '}';
    }

    

    
    
    
    
    
    
    
    
}

File: src/Entity/Produit.php
<?php

namespace App\Entity;

use App\Repository\ProduitRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;


/**
 * @ORM\Entity(repositoryClass=ProduitRepository::class)
 */
class Produit
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("produits")]
    private ?int $id_produit = null;

    /**
     * @ORM\Column(length=255)
     * @Assert\Length(min=4, minMessage="Veuillez entrer au minimum 4 caractères.")
     * @Assert\NotBlank(message="Ce champ ne doit pas être vide.")
     */
    #[ORM\Column]
    #[Groups("produits")]
    private ?string $nom_produit = null;

    /**
     * @ORM\Column(length=255)
     * @Assert\Length(min=4, minMessage="Veuillez entrer au minimum 4 caractères.")
     * @Assert\NotBlank(message="Ce champ ne doit pas être vide.")
     */
    #[ORM\Column]
    #[Groups("produits")]
    private ?string $description_produit = null;

    /**
     * @ORM\Column(type="float")
     * @Assert\GreaterThanOrEqual(value=0, message="Le prix ne peut pas être négatif.")
     */
    #[ORM\Column]
    #[Groups("produits")]
    private ?float $prix_produit = null;

    /**
     * @ORM\Column(length=255, nullable=true)
     */
    #[ORM\Column]
    #[Groups("produits")]
    private ?string $image_produit = null;

    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="produits")
     * @ORM\JoinColumn(name="id_categorie", referencedColumnName="id_categorie", nullable=false)
     */
    
    private ?Categorie $id_categorie = null;

   /**
     * @ORM\ManyToOne(targetEntity=User::class)
     * @ORM\JoinColumn(name="id_user", referencedColumnName="id_user")
     */
    
    private ?User $user = null;


    public function getIdProduit(): ?int
    {
        return $this->id_produit;
    }

    public function getNomProduit(): ?string
    {
        return $this->nom_produit;
    }

    public function setNomProduit(string $nom_produit): self
    {
        $this->nom_produit = $nom_produit;

        return $this;
    }

    public function getDescriptionProduit(): ?string
    {
        return $this->description_produit;
    }

    public function setDescriptionProduit(string $description_produit): self
    {
        $this->description_produit = $description_produit;

        return $this;
    }

    public function getPrixProduit(): ?float
    {
        return $this->prix_produit;
    }

    public function setPrixProduit(float $prix_produit): self
    {
        $this->prix_produit = $prix_produit;

        return $this;
    }

    public function getImageProduit(): ?string
    {
        return $this->image_produit;
    }

    public function setImageProduit(string $image_produit): self
    {
        $this->image_produit = $image_produit;

        return $this;
    }

    public function getIdCategorie(): ?Categorie
    {
        return $this->id_categorie;
    }

    public function setIdCategorie(?Categorie $id_categorie): self
    {
        $this->id_categorie = $id_categorie;

        return $this;
    }

    



   
    /**
 * @ORM\OneToMany(targetEntity=Favori::class, mappedBy="produit")
 */
private Collection $favoris;

    public function __construct()
    {
        $this->favoris = new ArrayCollection();
    }


    public function getNombreFavoris(): int
    {
        return $this->favoris->count();
    }



public function setIdUser(User $user): self
{
    $this->user = $user;
    return $this;
}

public function getIdUser(): ?User
{
    return $this->user;
}

/**
 * @return Collection<int, Favori>
 */
public function getFavoris(): Collection
{
    return $this->favoris;
}

public function addFavori(Favori $favori): self
{
    if (!$this->favoris->contains($favori)) {
        $this->favoris->add($favori);
        $favori->setProduit($this);
    }

    return $this;
}

public function removeFavori(Favori $favori): self
{
    if ($this->favoris->removeElement($favori)) {
        // set the owning side to null (unless already changed)
        if ($favori->getProduit() === $this) {
            $favori->setProduit(null);
        }
    }

    return $this;
}


public function isFavoriByUser(?User $user): bool
{
    if (!$user) {
        return false;
    }

    foreach ($this->favoris as $favori) {
        if ($favori->getUser() === $user) {
            return true;
        }
    }

    return false;
}









}
